File: TransCoder_tokenized_test_set_functions/python/PETERSON_GRAPH.py
def findthepath ( S, v ) :
    result [ 0 ] = v
    for i in range ( 1, len ( S ) ) :
        if ( adj [ v ] [ ord ( S [ i ] ) - ord ( 'A' ) ] or adj [ ord ( S [ i ] ) - ord ( 'A' ) ] [ v ] ) :
            v = ord ( S [ i ] ) - ord ( 'A' )
        elif ( adj [ v ] [ ord ( S [ i ] ) - ord ( 'A' ) + 5 ] or adj [ ord ( S [ i ] ) - ord ( 'A' ) + 5 ] [ v ] ) :
            v = ord ( S [ i ] ) - ord ( 'A' ) + 5
        else :
            return False
        result . append ( v )
    return True

File: TransCoder_tokenized_test_set_functions/cpp/PETERSON_GRAPH.cpp
bool findthepath ( char * S, int v ) {
  result [ 0 ] = v + '0';
  for ( int i = 1;
  S [ i ];
  i ++ ) {
    if ( adj [ v ] [ S [ i ] - 'A' ] || adj [ S [ i ] - 'A' ] [ v ] ) {
      v = S [ i ] - 'A';
    }
    else if ( adj [ v ] [ S [ i ] - 'A' + 5 ] || adj [ S [ i ] - 'A' + 5 ] [ v ] ) {
      v = S [ i ] - 'A' + 5;
    }
    else return false;
    result [ i ] = v + '0';
  }
  return true;
}